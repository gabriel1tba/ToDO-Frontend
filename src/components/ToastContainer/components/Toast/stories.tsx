import { Story, Meta } from '@storybook/react';

import Toast from '.';

import { IToast } from 'components/ToastContainer/interfaces';
import { v4 } from 'uuid';

export default {
  title: 'Toast',
  component: Toast,
} as Meta;

export const Default: Story<IToast> = (args) => <Toast {...args} />;

Default.args = {
  toastMessage: {
    id: v4(),
    title: 'Info',
    description: 'Info Description',
    secondsDuration: 4000,
  },
};

export const Success: Story<IToast> = (args) => <Toast {...args} />;

Success.args = {
  toastMessage: {
    id: v4(),
    type: 'success',
    title: 'Info',
    description: 'Info Description',
    secondsDuration: 4000,
  },
};

export const Error: Story<IToast> = (args) => <Toast {...args} />;

Error.args = {
  toastMessage: {
    id: v4(),
    type: 'error',
    title: 'Info',
    description: 'Info Description',
    secondsDuration: 4000,
  },
};
