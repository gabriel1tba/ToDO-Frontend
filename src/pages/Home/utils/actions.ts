import { TTodo } from 'services/TodoService/interfaces';

export enum Types {
  GET_TODOS = 'GET_TODOS',
  CREATE_TODO = 'CREATE_TODO',
  UPDATE_TODO = 'UPDATE_TODO',
  DELETE_TODO = 'DELETE_TODO',
}

interface GetTodos {
  type: Types.GET_TODOS;
  payload: TTodo[];
}

interface CreateTodo {
  type: Types.CREATE_TODO;
  payload: TTodo;
}

interface UpdateTodo {
  type: Types.UPDATE_TODO;
  payload: TTodo;
}

interface DeleteTodo {
  type: Types.DELETE_TODO;
  payload: string;
}

export type Action = GetTodos | CreateTodo | UpdateTodo | DeleteTodo;

export const getTodosAction = (todos: TTodo[]): GetTodos => ({
  type: Types.GET_TODOS,
  payload: todos,
});

export const createTodoAction = (todo: TTodo): CreateTodo => ({
  type: Types.CREATE_TODO,
  payload: todo,
});

export const updateTodoAction = (todo: TTodo): UpdateTodo => ({
  type: Types.UPDATE_TODO,
  payload: todo,
});

export const deleteTodoAction = (id: string): DeleteTodo => ({
  type: Types.DELETE_TODO,
  payload: id,
});
