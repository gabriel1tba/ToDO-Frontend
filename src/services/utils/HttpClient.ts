import axios from 'axios';

const HttpClient = axios.create({
  baseURL: `${process.env.REACT_APP_API_HOST}:${process.env.REACT_APP_API_PORT}`,
});

export default HttpClient;
