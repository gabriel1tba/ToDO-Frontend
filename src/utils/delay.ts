const delay = (ms = 1000): Promise<unknown> =>
  new Promise((resolve) => setTimeout(resolve, ms));

export default delay;
