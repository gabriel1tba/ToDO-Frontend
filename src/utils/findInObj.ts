// eslint-disable-next-line @typescript-eslint/no-explicit-any
const findInObj = (object: any, value: any): any => {
  // @OBSERVAÇÃO: Remove chaves com valores nulos
  // Object.keys(object).forEach(
  //   (key) => object[key] === null && delete object[key],
  // );

  return Object.values(object).some((v) =>
    typeof v === 'object' && !Array.isArray(object)
      ? findInObj(v, value)
      : typeof v === 'string'
      ? v.toLowerCase().includes(value.toLowerCase())
      : typeof v === 'number'
      ? String(v).includes(value) || (isNaN(v) && isNaN(value))
      : v === value,
  );
};

export default findInObj;
