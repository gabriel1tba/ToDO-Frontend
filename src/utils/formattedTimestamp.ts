export const formattedTimestamp = (timeStamp: string) => {
  return new Date(timeStamp).toLocaleString('pt-BR');
};

export default formattedTimestamp;
