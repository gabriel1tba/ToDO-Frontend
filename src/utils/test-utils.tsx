import { render, RenderOptions } from '@testing-library/react';
import { TodoProvider } from 'context/todos';

interface IWrapperProvider {
  children: React.ReactNode;
}

const WrapperProvider = ({ children }: IWrapperProvider) => (
  <TodoProvider>{children}</TodoProvider>
);

const customRender = (ui: React.ReactElement, options?: RenderOptions) =>
  render(<WrapperProvider>{ui}</WrapperProvider>, { ...options });

export * from '@testing-library/react';
export { customRender as render };
