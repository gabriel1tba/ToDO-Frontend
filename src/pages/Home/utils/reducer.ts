import { TTodo } from 'services/TodoService/interfaces';

import { Types, Action } from './actions';

const todosReducer = (state: TTodo[], action: Action): TTodo[] => {
  switch (action.type) {
    case Types.GET_TODOS: {
      return action.payload;
    }

    case Types.CREATE_TODO: {
      return [...state, action.payload];
    }

    case Types.UPDATE_TODO: {
      return state.map((todo) => {
        if (action.payload.id === todo.id) {
          todo = { ...action.payload, created_at: todo.created_at };
        }
        return todo;
      });
    }

    case Types.DELETE_TODO: {
      return state.filter((todo) => {
        return todo.id !== action.payload;
      });
    }

    default:
      return state;
  }
};

export default todosReducer;
