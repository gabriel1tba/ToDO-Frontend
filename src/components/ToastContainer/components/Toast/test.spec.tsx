import userEvent from '@testing-library/user-event';
import { render, screen } from 'utils/test-utils';

import Toast from '.';

const message = {
  id: '3c0e7f5d-02ed-4a97-9fe2-d901f7bbf275',
  title: 'Erro ao tentar logar!',
  description: 'Um erro inesperado aconteceu... Tente novamente mais tarde.',
  secondsDuration: 5,
};

describe('<Toast />', () => {
  it('should render the Toast correctly', () => {
    const { container } = render(<Toast style={{}} toastMessage={message} />);

    expect(screen.getByText(/erro ao tentar logar!/i)).toBeInTheDocument();
    expect(
      screen.getByText(
        /um erro inesperado aconteceu... Tente novamente mais tarde./i,
      ),
    ).toBeInTheDocument();

    expect(screen.getByRole('button')).toBeInTheDocument();

    expect(container.firstChild).toMatchSnapshot();
  });

  it('should close toast', () => {
    render(
      <Toast style={{}} toastMessage={{ ...message, secondsDuration: 0 }} />,
    );

    userEvent.click(screen.getByRole('button'));
  });
});
