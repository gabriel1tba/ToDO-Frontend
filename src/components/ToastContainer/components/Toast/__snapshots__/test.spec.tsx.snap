// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Toast /> should render the Toast correctly 1`] = `
.c0 {
  width: 320px;
  position: fixed;
  margin-right: 10px;
  padding: 16px 30px 16px 16px;
  border-radius: 10px;
  box-shadow: 2px 2px 8px rgba(0,0,0,0.2);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  background: #ebf8ff;
  color: #3172b7;
}

.c0 > svg {
  margin: 4px 12px 0 0;
}

.c0 div {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
}

.c0 div p {
  margin-top: 4px;
  font-size: 14px;
  opacity: 0.8;
  line-height: 20px;
}

.c0 button {
  cursor: pointer;
  position: absolute;
  right: 16px;
  top: 19px;
  opacity: 0.6;
  border: 0;
  background: transparent;
  color: inherit;
}

@media (max-width:375px) {
  .c0 {
    width: 280px;
    padding: 14px 26px 14px 14px;
  }
}

<div
  class="c0"
  hasdescription="1"
>
  <svg
    fill="none"
    height="24"
    stroke="currentColor"
    stroke-linecap="round"
    stroke-linejoin="round"
    stroke-width="2"
    viewBox="0 0 24 24"
    width="24"
    xmlns="http://www.w3.org/2000/svg"
  >
    <circle
      cx="12"
      cy="12"
      r="10"
    />
    <line
      x1="12"
      x2="12"
      y1="16"
      y2="12"
    />
    <line
      x1="12"
      x2="12.01"
      y1="8"
      y2="8"
    />
  </svg>
  <div>
    <strong>
      Erro ao tentar logar!
    </strong>
    <p>
      Um erro inesperado aconteceu... Tente novamente mais tarde.
    </p>
  </div>
  <button
    type="button"
  >
    <svg
      fill="none"
      height="18"
      stroke="currentColor"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="2"
      viewBox="0 0 24 24"
      width="18"
      xmlns="http://www.w3.org/2000/svg"
    >
      <circle
        cx="12"
        cy="12"
        r="10"
      />
      <line
        x1="15"
        x2="9"
        y1="9"
        y2="15"
      />
      <line
        x1="9"
        x2="15"
        y1="9"
        y2="15"
      />
    </svg>
  </button>
</div>
`;
